storage_service/locations/models/package.py

```diff
@@ -783,7 +785,10 @@ def check_fixity(self, delete_after=True):
         if self.is_compressed:
             # bagit can't deal with compressed files, so extract before
             # starting the fixity check.
-            path, temp_dir = self.extract_file()
+            try:
+                 path, temp_dir = self.extract_file()
+            except StorageException:
+                 return (False, [], 'Error extracting file')
```

Comment:
I'm a bit confused aboout the failures return parameter. It seems like it's supposed to be a list, but it gets set to the "details" property of BagValidationError which doesn't seem like it would be a list.